example/http/client/async-ssl/http_client_async_ssl.cpp

```diff
@@ -54,9 +54,9 @@ class session : public std::enable_shared_from_this<session>
     // Objects are constructed with a strand to
     // ensure that handlers do not execute concurrently.
     explicit
-    session(net::io_context& ioc, ssl::context& ctx)
-        : resolver_(net::make_strand(ioc))
-        , stream_(net::make_strand(ioc), ctx)
+    session(net::executor exec, ssl::context& ctx)
+    // note: defer to private constructor after having created the strand
+    : session(net::make_strand(exec), ctx)
```

Comment:
Anyway what is the necessity for being so verbose? Why not just call `make_strand` at the call site of `make_shared<session>`?

Reply:
Calling make_strand at the call site passes responsibility for thread-safety to the caller. I find this philosophically concerning as it's leaking information.

To be honest, the program does not actually require strands at all. There are no concurrent async operations in play.

I wonder whether we'd be better off removing all mention of strands altogether.

Reply:
> Calling make_strand at the call site passes responsibility for thread-safety to the caller.

Yes, that's exactly right. The caller is the one with the information on whether or not the strand is necessary, since only the caller knows how many threads are calling `io_context::run`.

> the program does not actually require strands at all.

It doesn't, but the philosophy of the examples is to always include them so that users have more robust code in the face of changes. 90% of crashing bug reports are caused by missing strands in multi-threaded contexts.

Reply:
If anything, we should either include a very brief comment on why we're using strands, and/or include a few sentences in the documentation about the use of strands in the examples. And we should audit every example and make sure they are using strands consistently. I agree that we want to transition from mentioning the `io_context` executor to the `strand` as early as possible.

Reply:
OK that makes sense.

To be more honest, I don't think this particular example requires a shared_ptr either, since there are no concurrent outstanding async operations and the lifetime of the session must necessarily end before the lifetime of the io_context.

Whether this example is the place to teach this is another matter.